dev/changelog/40.0.0.md

```diff
@@ -0,0 +1,372 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Apache DataFusion 40.0.0 Changelog
+
+This release consists of 261 commits from 64 contributors. See credits at the end of this changelog for more information.
+
+**Breaking changes:**
+
+- fix: remove the Sized requirement on ExecutionPlan::name() [#11047](https://github.com/apache/datafusion/pull/11047) (waynexia)
+- Return `&Arc` reference to inner trait object [#11103](https://github.com/apache/datafusion/pull/11103) (linhr)
+- remove `derive(Copy)` from `Operator` [#11132](https://github.com/apache/datafusion/pull/11132) (samuelcolvin)
```

Comment:
since we reverted this issue it probably isn't a breaking change

Reply:
🤔  it might be good to list the other PRs related to porting aggregate UDFs here too

I tagged a few more PRs with breaking change in case you happen to rerun the changelog generator